### ISSUE TITLE
Unit-less values

### ISSUE THREAD

**Issue by Granipouss** [josdejong/mathjs#1240]
Hi,

When I try to execute `mathjs.unit('5')`, it raises an error `Uncaught SyntaxError: "5" contains no units` when it should simply create a a variable with value 5 and all units set to 0. I simple work around is to execute `mathjs.unit('5 s/s')` instead, but I still think this is an issue that can be fixed in a cleaner manner.

Cheers,

**Comment by ericman314** [comment]
I agree. It seems like a trivial case but one that we could easily handle.

**Comment by josdejong** [comment]
Should indeed be easy to fix. Just curious, what is your use case for `mathjs.unit('5')` @Granipouss?

**Comment by Granipouss** [comment]
I am working on a nutrition app and I am using mathjs to handle all the rate computation. Food can be measured by mass (for flour, etc.), volume (milk, etc.) or unit (eggs, etc.). I can use all my formula with the proper units for mass and volume but I have to do a workaround for the unit based food item.

**Comment by josdejong** [comment]
@Granipouss thanks that helps. I guess I'm overlooking something, but why can't you use just the number `5` in your use case?

**Comment by Granipouss** [comment]
I can but the thing is that then quantities are not homogeneous, sometimes it is a `mathjs.unit` (whether is it kg or L), sometimes is a native `Number`. And if I want to display them for instance, I have to use `format` or `toString` from the former or nothing for the later.

**Comment by josdejong** [comment]
Ah, I understand that. If you want to use methods on the units, you need to make sure you're always dealing with a unit. 

You could use functions instead like `math.format(unit)` instead of methods like `unit.format()`. The functions `math.format()` and `math.string()` work with any input value, like units and numbers.

I don't think it hurts though to allow creating a unit-less unit, we can add it.

**Comment by ericman314** [comment]
I've submitted #1251 which I think should address this issue. Despite all tests passing, I don't know yet whether there will be any regressions caused by allowing the creation of a dimensionless unit. I don't think there will be, since `math.unit('5 s/s')` seems to work just fine.

**Comment by josdejong** [comment]
This is available now in `v5.2.0`, thanks again @ericman314 . And sorry for the delay, I had totally forgotten to release this improvement earlier.

**Comment by dpnova** [comment]
Old issue, but it seems that the reviver or something doesn't like reviving unitless objects, drop this into the repl [linked](https://jsbin.com/hiwexatibo/edit?html,output) from the homepage:

```
    print(
      JSON.parse(
        JSON.stringify(math.unit("5"), math.replacer), 
        math.reviver
      )
    );
```

**Comment by josdejong** [comment]
Thanks for reporting David. Fixed via da0c70ebe39a6b2581315ac617a21f11d76e2cee (not yet published)

**Comment by josdejong** [comment]
The fix is published now in `v12.4.3`
